lib/rules/no-unused-vars.js

```diff
@@ -74,6 +74,26 @@ module.exports = function(context) {
         return [].concat.apply(unused, [].map.call(scope.childScopes, unusedLocals));
     }
 
+    function flagJSXAsUsed(nodeName) {
```

Comment:
I suggest `flagVariableAsUsedInJSX` to make it clear that we set a flag on a variable.